src/editor/Editor.js

```diff
@@ -2385,6 +2401,133 @@ define(function (require, exports, module) {
         }
     };
 
+    /**
+     * Clears all marks from the gutter with the specified name.
+     * @param {string} name The name of the gutter to clear.
+     */
+    Editor.prototype.clearGutter = function (name) {
+        this._codeMirror.clearGutter(name);
+    };
+
+    /**
+     * Renders all registered gutters
+     * @private
+     */
+    Editor.prototype._renderGutters = function () {
+        var languageId = this.document.getLanguage().getId();
+
+        function _filterByLanguages(gutter) {
+            return !gutter.languages || gutter.languages.indexOf(languageId) > -1;
+        }
+
+        function _sortByPriority(a, b) {
+            return a.priority - b.priority;
+        }
+
+        function _getName(gutter) {
+            return gutter.name;
+        }
+
+        var gutters = registeredGutters.map(_getName),
+            rootElement = this.getRootElement();
+
+        // If the line numbers gutter has not been explicitly registered and the CodeMirror lineNumbes option is
+        // set to true, we explicitly add the line  numbers gutter. This case occurs the first time the editor loads
+        // and showwLineNumbers is set to true in preferences
+        if (gutters.indexOf(LINE_NUMBER_GUTTER) < 0 && this._codeMirror.getOption(cmOptions[SHOW_LINE_NUMBERS])) {
+            registeredGutters.push({name: LINE_NUMBER_GUTTER, priority: LINE_NUMBER_GUTTER_PRIORITY});
+        }
+
+        gutters = registeredGutters.sort(_sortByPriority)
+            .filter(_filterByLanguages)
+            .map(_getName);
+
+        this._codeMirror.setOption("gutters", gutters);
+        this._codeMirror.refresh();
+
+        if (gutters.indexOf(LINE_NUMBER_GUTTER) < 0) {
+            $(rootElement).addClass("linenumber-disabled");
+        } else {
+            $(rootElement).removeClass("linenumber-disabled");
+        }
+    };
+
+    /**
+     * Sets the marker for the specified gutter on the specified line number
+     * @param   {string}   lineNumber The line number for the inserted gutter marker
+     * @param   {string}   gutterName The name of the gutter
+     * @param   {object}   marker     The jquery object representint the marker to the inserted in the gutter
+     */
+    Editor.prototype.setGutterMarker = function (lineNumber, gutterName, marker) {
+        var gutterNameRegistered = registeredGutters.some(function (gutter) {
+            return gutter.name === gutterName;
+        });
+
+        if (!gutterNameRegistered) {
+            console.warn("Gutter name must be registered before calling Editor.setGutterMarker");
+            return;
+        }
+
+        this._codeMirror.setGutterMarker(lineNumber, gutterName, marker);
+    };
+
+    /**
+     * Returns the list of gutters current registered on this editor.
```

Comment:
I think now the registered gutters are shared among all editors, correct? If so the description has to be updated.